code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -68,6 +71,9 @@ obj/item/gun/ballistic/rifle/attackby(obj/item/A, mob/user, params)
 	if(.)
 		spread = 36
 		can_bayonet = FALSE
+		worn_icon_state = "gun"
+		worn_icon = 'icons/mob/clothing/belt.dmi'
```

Comment:
already handled by the base proc.

Reply:
Reverted. The updated sawoff base proc fixes worn_icon_state. Worn_icon omitted.